File: examples-sdk/go/029-basic-vil-handler-endpoint/main.go
// 029-basic-vil-handler-endpoint — Go SDK equivalent
// Compile: vil compile --from go --input 029-basic-vil-handler-endpoint/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("macro-demo", 8080)

	demo := vil.NewService("demo")
	demo.Endpoint("GET", "/plain", "plain_handler")
	demo.Endpoint("GET", "/handled", "handled_handler")
	demo.Endpoint("POST", "/endpoint", "endpoint_handler")
	s.Service(demo)

	s.Compile()
}

File: examples-sdk/zig/029-basic-vil-handler-endpoint/main.zig
// 029-basic-vil-handler-endpoint — Zig SDK equivalent
// Compile: vil compile --from zig --input 029-basic-vil-handler-endpoint/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("macro-demo", 8080);
    var demo = vil.Service.init("demo");
    demo.endpoint("GET", "/plain", "plain_handler");
    demo.endpoint("GET", "/handled", "handled_handler");
    demo.endpoint("POST", "/endpoint", "endpoint_handler");
    server.service(&demo);
    server.compile();
}
